### ISSUE TITLE
styled-icons is 100MB

### ISSUE THREAD

**Issue by znarf** [opencollective/opencollective#1832]
**Describe the bug**

`styled-icons` is taking 100MB and is a main dependency, so not  pruned in production.

Would be great if we didn't have to ship that in our build.

```
-----> Compressing...
       Done: 307.7M
-----> Launching...
 !     Warning: Your slug size exceeds our soft limit (307 MB) which may affect boot time.
```

![Screen Shot 2019-03-19 at 20 44 34](https://user-images.githubusercontent.com/806/54637187-4d09b400-4a88-11e9-8f7c-7469f22f4cb5.png)

**Comment by Betree** [comment]
One way to fix that would be to have styled-icons exported as individual packages.
Eg. `@styled-icons/icomoon`, `@styled-icons/font-awesome`...etc

So we would import only the libraries that what we need.

Should I open an issue on styled-icons to suggest this?

**Comment by znarf** [comment]
@Betree In the end, I think `styled-icons` should be a dev dependency and pruned in production. Once bundled, we don't need the `node_modules/styled-icons` folder anymore.

**Comment by znarf** [comment]
Still valid.

**Comment by znarf** [comment]
A $50 bounty was attached to this issue. Anyone submitting a Pull Request will be rewarded with $50 when the Pull Request is reviewed, accepted and merged. [More info](BOUNTY.md)

This bounty is currently reserved to participants of the OSCA (Open Source Africa) event. If not completed there, it will be later open to everyone.

**Comment by debemenitammy** [comment]
Hi I'm Deborah from the OSCA Event Port Harcourt and I'll be working on this issue.

**Comment by debemenitammy** [comment]
> One way to fix that would be to have styled-icons exported as individual packages.
> Eg. @styled-icons/icomoon, @styled-icons/font-awesome...etc

Should I do it this way?

**Comment by znarf** [comment]
@debemenitammy Try to move the package to `devDependencies` and see if that works. You might have to tweak eslint warnings for each import.

**Comment by ernestmoi** [comment]
@znarf  Hi, i am Omakoji from OSCA Lagos i was able to move the package to devDependencies and it works what do i do next?

**Comment by znarf** [comment]
@ernestmoi Feel free to submit a pull request, we will review it and discuss it there.

**Comment by jamesgeorge007** [comment]
@znarf I would love to give it a shot :+1:

**Comment by znarf** [comment]
@jamesgeorge007 have a try but it might be a bit though to figure the actual solution. 

- it needs to fix the problem: have a smaller `node_modules` in production
- while not introducing too much complexity or warnings

The "moving to devDependencies" approach didn't work when someone tried. See: https://github.com/opencollective/opencollective-frontend/pull/2545

**Comment by someofthethings** [comment]
Sorry to be a bit curveball on this an think it terms of general industry experience and not your repo, but basic question: what is styled-icons giving you? Could I just write it myself with a lot less code?

**Comment by Betree** [comment]
@someofthethings `styled-icons` wrap multiple icons libraries and bundle them in react-components so we don't have to import them all. We're pretty happy with it and don't want to maintain such feature ourselves.

**Comment by someofthethings** [comment]
@Betree ah, but people say they are pretty happy with Passport... ;)

Keep on keepin' on tho!

**Comment by sbinlondon** [comment]
I was having a look at this one last night and might do some research into it if that's something we still need?

**Comment by jzabala** [comment]
I think @Betree [solution](https://github.com/opencollective/opencollective/issues/1832#issuecomment-474639542) would be the perfect one for this issue but in case we want this faster we could also create our own implementation of the library. 

We could create @opencollective/styled-icon. In this implementation we import the real one, export only what we need, use webpack tree shaking magic to build only the exported modules and publish that to npm.

**Comment by znarf** [comment]
I don't think we want at this point to move forward with a complex solution and maintain our own styled-icon library. If there is no easy fix, then we can live with the issue, it doesn't have much impact on our performance at this point.

Removing the bounty to give less exposure to the issue from now.

**Comment by Betree** [comment]
Opened [an issue](https://github.com/jacobwgillespie/styled-icons/issues/1055) in `styled-icons` to suggest exporting individual packages.

**Comment by jacobwgillespie** [comment]
Just posting an update here, I'm working on this at https://github.com/jacobwgillespie/styled-icons/pull/1056.  Still a work-in-progress, but I should have individual icon packages published shortly.

**Comment by jacobwgillespie** [comment]
I have released this as v9.0.0 - can you test here and let me know if you run into any issues?  You can essentially replace `styled-icons/PACK` with `@styled-icons/PACK`, I'll update the README shortly.

**Comment by jacobwgillespie** [comment]
I may have an issue to look into actually, it seems it's having trouble resolving dependencies - I'm off to bed, but will check this out further tomorrow.

**Comment by jeffin143** [comment]
Great job @jacobwgillespie :rocket: :fire: 

That was really a great improvement - almost 3 Times the reduced size 

Made a mock test - Here are the results 

**v8 - 106 MB**

![Screenshot from 2019-11-22 10-06-00](https://user-images.githubusercontent.com/21286134/69398391-0c95c080-0d10-11ea-8b21-0fc4d857f1b9.png)

**v9 - 36 MB**

![Screenshot from 2019-11-22 10-06-33](https://user-images.githubusercontent.com/21286134/69398392-0c95c080-0d10-11ea-916a-d98263735393.png)

**Comment by Betree** [comment]
You're amazing @jacobwgillespie, thanks for the super-fast action on this :open_mouth:

**Comment by znarf** [comment]
@jeffin143 fancy sharing a PR for it?

**Comment by jacobwgillespie** [comment]
I think I fixed the dependency issue in v9.0.1

@jeffin143 note that you're still using `styled-icons`, which means all the icons are still included, they just moved into the `@styled-icons` directory (one left of the one you highlighted in that screenshot).  To actually get a size reduction, you'll want to only install `@styled-icons/PACK` modules for each icon pack in use here.

**Comment by jeffin143** [comment]
> @jeffin143 fancy sharing a PR for it?

I will do it
